stackit/internal/services/loadbalancer/credential/resource.go

```diff
@@ -0,0 +1,353 @@
+	}
+
+	var apiClient *loadbalancer.APIClient
+	var err error
+	if providerData.LoadBalancerCustomEndpoint != "" {
+		ctx = tflog.SetField(ctx, "loadbalancer_custom_endpoint", providerData.LoadBalancerCustomEndpoint)
+		apiClient, err = loadbalancer.NewAPIClient(
+			config.WithCustomAuth(providerData.RoundTripper),
+			config.WithEndpoint(providerData.LoadBalancerCustomEndpoint),
+		)
+	} else {
+		apiClient, err = loadbalancer.NewAPIClient(
+			config.WithCustomAuth(providerData.RoundTripper),
+			config.WithRegion(providerData.Region),
+		)
+	}
+
+	if err != nil {
+		core.LogAndAddError(ctx, &resp.Diagnostics, "Error configuring API client", fmt.Sprintf("Configuring client: %v. This is an error related to the provider configuration, not to the resource configuration", err))
+		return
+	}
+
+	r.client = apiClient
+	tflog.Info(ctx, "Load Balancer client configured")
+}
+
+// Schema defines the schema for the resource.
+func (r *credentialResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	descriptions := map[string]string{
+		"main":            "Load balancer credential resource schema. Must have a `region` specified in the provider configuration.",
+		"id":              "Terraform's internal resource ID. It is structured as \"`project_id`\",\"`credentials_ref`\".",
+		"project_id":      "STACKIT project ID to which the load balancer credential is associated.",
+		"display_name":    "Credential name.",
+		"username":        "The username used for the ARGUS instance.",
+		"password":        "The password used for the ARGUS instance.",
+		"credentials_ref": "The credentials reference can be used for observability of the Load Balancer.",
+	}
+
+	resp.Schema = schema.Schema{
+		Description: descriptions["main"],
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: descriptions["id"],
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"project_id": schema.StringAttribute{
+				Description: descriptions["project_id"],
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					validate.UUID(),
+				},
+			},
+			"display_name": schema.StringAttribute{
+				Description: descriptions["display_name"],
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"username": schema.StringAttribute{
+				Description: descriptions["username"],
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"password": schema.StringAttribute{
+				Description: descriptions["password"],
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"credentials_ref": schema.StringAttribute{
```

Comment:
Since this is doing what `[RESOURCE]_id` does usually,
- Nitpick: put it after the `id` field in the schema, so the identifiers are at the top
- Does it make sense to call it `credential_id`, for uniformity with other resources? Or should we prefer uniformity with the API? Not sure...

Reply:
1. I'll move it to the top
2. Good point. But I think generally when we have `...Id` the field is a UUID and in this case it's not, and for some resources the identifier is a `name`

Reply:
Moved it to the top but it didn't produce any difference in the docs (it's ordered alphabetically)